db_sql.go

```diff
@@ -469,89 +580,161 @@ func SqlCreateDatabase(conx *DBConnection) error {
+			// Tracking tables dont exist, this is up to date
+			return true, nil
+		}
+		return false, err
+	}
 
-			for ; version < len(migrations); version += 1 {
-				fmt.Printf("Migrating %v to version %v\n", schema_name, version+1)
-				if _, err := tx.Exec(migrations[version]); err != nil {
-					panic(err)
-				}
-				if _, err := tx.Exec(fmt.Sprintf("UPDATE %v_version SET version = %v", schema_name, version+1)); err != nil {
-					panic(err)
-				}
-			}
-			return nil
+	getTableVersion := func(table string) (int, error) {
+		var version int
+		row := db.QueryRow(fmt.Sprintf("SELECT version FROM %v", table))
+		err := row.Scan(&version)
+		if err != nil {
+			return -1, err
 		} else {
-			// Match the string 'database "foo" does not exist' to detect when the database needs to be created
-			if tryCreatingDB && strings.Index(eTxBegin.Error(), "does not exist") != -1 {
-				if eCreateDB := SqlCreateDatabase(conx); eCreateDB != nil {
-					return eCreateDB
-				} else {
-					return migrateSchemaInternal(false, conx, schema_name, migrations)
-				}
-			} else {
-				return eTxBegin
-			}
+			return version, nil
 		}
 	}
-}
-
-// Create a Postgres DB schema necessary for a Session database
-func SqlCreateSchema_Session(conx *DBConnection) error {
-	versions := make([]string, 0)
-	versions = append(versions, `
-	CREATE TABLE authsession (id BIGSERIAL PRIMARY KEY, sessionkey VARCHAR, identity VARCHAR, permit VARCHAR, expires TIMESTAMP);
-	CREATE UNIQUE INDEX idx_authsession_token    ON authsession (sessionkey);
-	CREATE        INDEX idx_authsession_identity ON authsession (identity);
-	CREATE        INDEX idx_authsession_expires  ON authsession (expires);`)
-	// we made some changes that are not backwards compatible, need to force a logout for all users
-	versions = append(versions, `DELETE FROM authsession;`)
-
-	return MigrateSchema(conx, "authsession", versions)
-}
 
-// Create a Postgres DB schema suitable for storage of Permits and Authentication
-// Note that we COULD separate Permit and Authenticator, but I'm not sure that provides any value.
-// It would be trivial to do so if that need ever arises.
-func SqlCreateSchema_User(conx *DBConnection) error {
-	versions := make([]string, 0)
-	versions = append(versions, `
-	CREATE TABLE authuser (id BIGSERIAL PRIMARY KEY, identity VARCHAR, password VARCHAR, permit VARCHAR);
-	CREATE UNIQUE INDEX idx_authuser_identity ON authuser (identity);`)
+	version1, err1 := getTableVersion(oldVersionTables[0])
+	if err1 != nil {
+		return false, err1
+	}
+	version2, err2 := getTableVersion(oldVersionTables[1])
+	if err2 != nil {
+		return false, err2
+	}
+	version3, err3 := getTableVersion(oldVersionTables[2])
+	if err3 != nil {
+		return false, err3
+	}
 
-	versions = append(versions, `
-	DROP INDEX idx_authuser_identity;
-	CREATE UNIQUE INDEX idx_authuser_identity ON authuser (LOWER(identity));`)
+	if version1 == 1 && version2 == 2 && version3 == 3 {
```

Comment:
There should be a comment here explaining the magic constants 1, 2, 3. To the casual observer, it looks strange that they're perfectly correlated with version1, version2, version3